nxp_s32k/libuavcan/driver/include/s32k_libuavcan.hpp

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2020, NXP. All rights reserved.
+ * Distributed under The MIT License.
+ * Author: Abraham Rodriguez <abraham.rodriguez@nxp.com>
+ */
+
+/**
+ * @file
+ * Header driver file for the media layer of Libuavcan v1 targeting
+ * the NXP S32K14 family of automotive grade MCU's running
+ * CAN-FD at 4Mbit/s data phase and 1Mbit/s in nominal phase.
+ */
+
+#ifndef S32K_LIBUAVCAN_HPP_INCLUDED
+#define S32K_LIBUAVCAN_HPP_INCLUDED
+
+/** Driver build configurations file */
+#include "build_config.hpp"
+
+/** libuavcan core header files */
+#include "libuavcan/media/can.hpp"
+#include "libuavcan/media/interfaces.hpp"
+#include "libuavcan/platform/memory.hpp"
+
+/** STL queue for the intermediate ISR buffer */
+#include <deque>
+
+/** CMSIS Core for __REV macro use */
+#include "s32_core_cm4.h"
+
+/**
+ * Preprocessor conditionals for deducing the number of CANFD FlexCAN instances in target MCU,
+ * this macro is defined inside the desired memory map "S32K14x.h" included header file
+ */
+#if defined(MCU_S32K142) || defined(MCU_S32K144)
+#    define TARGET_S32K_CANFD_COUNT (1u)
+#    define DISCARD_COUNT_ARRAY 0
+
+#elif defined(MCU_S32K146)
+#    define TARGET_S32K_CANFD_COUNT (2u)
+#    define DISCARD_COUNT_ARRAY 0, 0
+
+#elif defined(MCU_S32K148)
+#    define TARGET_S32K_CANFD_COUNT (3u)
+#    define DISCARD_COUNT_ARRAY 0, 0, 0
+
+#else
+#    error "No NXP S32K compatible MCU header file included"
+#endif
+
+namespace libuavcan
+{
+namespace media
+{
+/**
+ * @namespace S32K
+ * Microcontroller-specific constants, variables and non-mutating helper functions for the use of the FlexCAN peripheral
+ */
+namespace S32K
```

Comment:
These aren't needed by consumers of the driver and should be in the cpp file.

Reply:
Fixed